actionpack/test/controller/integration_test.rb

```diff
@@ -1126,3 +1126,46 @@ def test_cookies_set_in_setup_are_persisted_through_the_session
     assert_equal({"user_name"=>"david"}, cookies.to_hash)
   end
 end
+
+class IntegrationRequestEncodersTest < ActionDispatch::IntegrationTest
+  class FooController < ActionController::Base
+    def foos
+      render plain: 'ok'
+    end
+  end
+
+  def test_encoding_as_json
+    assert_encoded_as :json, content_type: 'application/json'
+  end
+
+  def test_encoding_as_without_mime_registration
+    assert_raise ArgumentError do
+      ActionDispatch::IntegrationTest.register_encoder :wibble
+    end
+  end
+
+  def test_registering_custom_encoder
+    Mime::Type.register 'text/wibble', :wibble
+
+    ActionDispatch::IntegrationTest.register_encoder(:wibble, &:itself)
+
+    assert_encoded_as :wibble, content_type: 'text/wibble',
+      parsed_parameters: Hash.new # Unregistered MIME Type can't be parsed
```

Comment:
@rafaelfranca tried your suggestion to call a custom controller via curl, but couldn't get it to work.

When I byebug `request_parameters` it seems to `fetch_header` which calls `@env.fetch` which already seems to have an empty hash for that key. Would you have any ideas why? I'll keep digging :grin: